nav2_core/include/nav2_core/local_planner_base.hpp

```diff
@@ -0,0 +1,119 @@
+ *
+ *  Copyright (c) 2017, Locus Robotics
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions
+ *  are met:
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   * Neither the name of the copyright holder nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef NAV2_CORE__LOCAL_PLANNER_BASE_HPP_
+#define NAV2_CORE__LOCAL_PLANNER_BASE_HPP_
+
+#include <memory>
+#include <string>
+
+#include "nav2_costmap_2d/costmap_2d_ros.hpp"
+#include "tf2_ros/transform_listener.h"
+#include "nav2_util/lifecycle_node.hpp"
+#include "pluginlib/class_loader.hpp"
+
+#include "nav_2d_msgs/msg/pose2_d_stamped.hpp"
+#include "nav_2d_msgs/msg/twist2_d_stamped.hpp"
+#include "nav_2d_msgs/msg/path2_d.hpp"
+
+namespace nav2_core
+{
+
+/**
+ * @class LocalPlanner
+ * @brief planner interface that will be inherited by all local planners
+ */
+class LocalPlanner : public nav2_util::LifecycleHelperInterface
+{
+public:
+  using Ptr = std::shared_ptr<nav2_core::LocalPlanner>;
+
+  /**
+   * @brief Constructor that brings up pluginlib loaders
+   */
+
+  virtual void initialize(nav2_util::LifecycleNode::SharedPtr,
+    std::shared_ptr<nav2_costmap_2d::Costmap2DROS>) = 0;
+  /**
+   * @brief Virtual destructor
+   */
+  virtual ~LocalPlanner() {}
+
+  /**
+   * @brief Sets the global goal for this local planner.
+   * @param goal_pose The Goal Pose
+   */
+
+  // virtual void setGoalPose(const nav_2d_msgs::msg::Pose2DStamped& goal_pose) = 0;
+
+  /**
+   * @brief local setPlan - Sets the global plan
+   * @param path The global plan
+   */
+  virtual void setPlan(const nav_2d_msgs::msg::Path2D & path) = 0;
```

Comment:
Do me a favor and change these all to use `nav2_msgs` or standard 3D enabled messages. The first thing you do in setPlan/computeVelocityCommands can just be converting back but that way its flexible. Given there's work happening to port other local planners, this is the kind of change we should make right away so we're not breaking `N` packages later. We have a little more leeway on the other headers since to my knowledge no one has written any substantial recovery plugins to break/port

Reply:
Done.